iceoryx_hoofs/include/iceoryx_hoofs/cxx/stack.hpp

```diff
@@ -50,15 +50,15 @@ class stack // NOLINT (cppcoreguidelines-pro-type-member-init,hicpp-member-init)
     static constexpr uint64_t capacity() noexcept;
 
   private:
-    // AXIVION Next Line AutosarC++19_03-A18.1.1 : safe access is guaranteed since the char array is wrapped inside the
-    // stack class
+    // AXIVION Next Construct AutosarC++19_03-A18.1.1 : safe access is guaranteed since the char array is wrapped inside
+    // the stack class
     /// @NOLINTNEXTLINE(hicpp-avoid-c-arrays, cppcoreguidelines-avoid-c-arrays)
     using element_t = uint8_t[sizeof(T)];
-    // AXIVION Next Line AutosarC++19_03-A18.1.1 : safe access is guaranteed since the char array is wrapped inside the
-    // stack class
+    // AXIVION Next Construct AutosarC++19_03-A18.1.1 : safe access is guaranteed since the char array is wrapped inside
```

Comment:
@MatthiasKillat good point, `std::aligned_storage_t` does half of what we want to achieve with `cxx::uninitialized_array`. This could be a good intermediate step. It seems it will be deprecated with C++23 though.

Reply:
@FerdinandSpitzschnueffler Surely the `alignas(T) element_t m_data[Capacity];` definition is confusing the static analysis and there are intricate details that may lead to subtle bugs (or so I think).

I was first mistaken on alignment handling, then I rethought it, was mistaken again ... and now I think I have all pieces put together:

1. `element_t` has alignment restriction 1 (raw byte array) and size `sizeof(T)` (so a T fits there)
2. `sizeof(T)` is always a multiple of `alignof(T)` (see comment where the C++ standard is referenced)
3. `m_data` is aligned as T

i) From all of them it follows that all elements of `m_data` are aligned as T (starting address is multiple of alignment due to 3. and every subsequent element is a multiple due since we add `k*sizeof(T)` and 2.)
ii) From 1. it follows that there is sufficient space for a T at any array element

Conclusion: it is not broken (wrt. size or alignment of data), but the static code analysis is not able to conclude the above and hence reports warnings. The warnings could maybe be reduced if `element_t` is aliged, but functionally it changes nothing

Reply:
@MatthiasKillat Thanks for the detailed explanation. Since I cannot reproduce the warning anymore, I remove the corresponding changes for now. We will see later if we have to align `element_t` or even implement the uninitialized array.

Reply:
@FerdinandSpitzschnueffler could you nevertheless move the alignment to `element_t`? It is still the correct way to do it, even if it works now. Let's assume someone adds another method and for whatever reason `element_t` is created on the stack. It is not visible that the alignment of `element_t` will be incorrect and a bug is introduced.

Reply:
@elBoberido this is not possible you cannot provide an alignment for a type alias. The simplest solution to add an alignment for `element_t` would be to introduce a struct of the correct size and alignment:
```cxx
template <typename T>
struct alignas(T) element_t {
    uint8_t dontTouchMe[sizeof(T)];
};
```

But this shouldn't complicate things since we will never access `dontTouchMe` ... aaah the old dont touch me ... never was a variable so often touched as the dont touch me ...

Reply:
@elfenpiff okay, I forgot that it was a alias and not a struct. We desperately need the `uninitialized_array`

Reply:
@elBoberido I think the `uninitialized_array` could be quickly implemented by just using the struct from above and add the method `T& operator[](const uint64_t id) noexcept;`.
Then we store the `reinterpret_cast` in there and it is done.

Maybe we can do this also in this PR and see how it feels. This would mean 3 additional files (header, inl and maybe simplistic test) but I would be alright with since we see the thing in action right here.